src/util/Helpers.cpp

```diff
@@ -316,6 +322,50 @@ QLocale getSystemLocale()
     return QLocale::system();
 }
 
+void writeProviderEmotesCache(const QString &id, const QString &provider,
+                              const QByteArray &bytes)
+{
+    QThreadPool::globalInstance()->start([bytes, id, provider]() {
+        QFile responseCache(getApp()->getPaths().cacheDirectory() + "/" + id +
+                            "." + provider);
+
+        if (responseCache.open(QIODevice::WriteOnly))
+        {
+            qCDebug(chatterinoCache)
+                << "Saved json response " << id << "." << provider;
+            responseCache.write(qCompress(bytes));
+        }
+    });
```

Comment:
is writing to a file really so slow that this needs to be done on a separate thread?

Reply:
In my opinion, yes. For example I have an external 10 TB HDD, which if it hasn't been used a while needs time to be responsive again. If the drive, the file is located on, is being used a lot anyways I don't see it as a big problem, tho.

Reply:
nymn's 7TV channel emotes, even compressed, is 589KB

some of it has to do with how large the network payload is since we're not only storing raw emote id/url stuff, but still kinda large